### DISCUSSION TITLE
How to Open WinUI 3 App from deep link?

### DISCUSSION THREAD

**Discussion by naarsi** [microsoft/microsoft-ui-xaml#9362] [Q&A]
I'm running into "Get an app to open this link" error when I try to open my app via deep link.
I have added the uap:Extension in my Application section of Package.appxmanifest: 

```
<Applications>
  <Application Id="myapptest"
    Executable="$targetnametoken$.exe"
    EntryPoint="$targetentrypoint$">
    <uap:VisualElements
      DisplayName="ms-resource:AppDisplayName"
      Description="ms-resource:AppDescription"
      BackgroundColor="transparent"
      Square150x150Logo="Assets\Square150x150Logo.png"
      Square44x44Logo="Assets\Square44x44Logo.png">
      <uap:DefaultTile Wide310x150Logo="Assets\Wide310x150Logo.png" />
      <uap:SplashScreen Image="Assets\SplashScreen.png" />
    </uap:VisualElements>
    <Extensions>
        <desktop:Extension Category="windows.toastNotificationActivation">
            <desktop:ToastNotificationActivation ToastActivatorCLSID="b3331ea8-5355-4335-8b5a-58f96bdf3b59" />
        </desktop:Extension>

        <com:Extension Category="windows.comServer">
            <com:ComServer>
                <com:ExeServer Executable="myapptest.exe" Arguments="----AppNotificationActivated:" DisplayName="Toast activator">
                    <com:Class Id="b3331ea8-5355-4335-8b5a-58f96bdf3b59" DisplayName="Toast activator"/>
                </com:ExeServer>
            </com:ComServer>
        </com:Extension>
  <uap:Extension Category="windows.protocol">
	  <uap:Protocol Name="myapptest"/>
  </uap:Extension>
    </Extensions>
  </Application>
</Applications>
```
Am I missing anything?

WindowsAppSDK package version: 1.4.231008000

**Comment by marcelwgn** [comment]
In my testing, creating a new app and just adding the protocol activation worked. Note though, the app won't be activated if it's not published and installed as a package since otherwise the registration might not work. Looking at your appxmanifest, I don't see any errors (assuming the com and desktop namespaces are correct). Did you make any modifications to the App.xaml.cs file? Are you able to reproduce this with a blank project?

**Reply by naarsi** [reply]
> Did you make any modifications to the App.xaml.cs file?

No changes that would impact this functionality, just registering the services. I am setting DISABLE_XAML_GENERATED_MAIN in my project tho, will that cause any issues?
 
I am publishing an my app via [Velopack](https://github.com/velopack/velopack), I can see the app in Settings Installed Apps section. 

> Are you able to reproduce this with a blank project?

Not yet, will try this.
